src/Agent.Worker/JobRunner.cs

```diff
@@ -250,6 +253,50 @@ public sealed class JobRunner : AgentService, IJobRunner
             }
         }
 
+        private async Task<TaskResult> CompleteJobAsync(IJobServer jobServer, IExecutionContext jobContext, AgentJobRequestMessage message, TaskResult? taskResult = null)
+        {
+            var result = jobContext.Complete(taskResult);
+            var outputVariables = jobContext.Variables.GetOutputVariables();
+            var webApiVariables = outputVariables.ToWebApiVariables();
+            var jobCompletedEvent = new JobCompletedEvent(message.RequestId, message.JobId, result, webApiVariables);
+
+            var completeJobRetryLimit = 5;
+            var exceptions = new List<Exception>();
+            while (completeJobRetryLimit-- > 0)
```

Comment:
There is a RetryExecutor but this is under Agent.Worker\Release. Can you move up the RetryExecutor and use it here?